src/Microsoft.IdentityModel.Tokens/InMemoryCryptoProviderCache.cs

```diff
@@ -26,20 +26,60 @@
 //------------------------------------------------------------------------------
 
 using System;
-using System.Collections.Concurrent;
 using System.Globalization;
 using Microsoft.IdentityModel.Logging;
+#if NET461 || NET472 || NETSTANDARD2_0 
+using Microsoft.Extensions.Caching.Memory;
+using Microsoft.Extensions.Options;
+#endif
 
 namespace Microsoft.IdentityModel.Tokens
 {
     /// <summary>
     /// Defines a cache for crypto providers.
     /// Current support is limited to <see cref="SignatureProvider"/> only.
     /// </summary>
-    public class InMemoryCryptoProviderCache : CryptoProviderCache
+#if NET461 || NET472 || NETSTANDARD2_0 
+    public class InMemoryCryptoProviderCache : CryptoProviderCache, IDisposable
+#elif NET45
+    public class InMemoryCryptoProviderCache: CryptoProviderCache
+#endif
     {
-        private ConcurrentDictionary<string, SignatureProvider> _signingSignatureProviders = new ConcurrentDictionary<string, SignatureProvider>();
-        private ConcurrentDictionary<string, SignatureProvider> _verifyingSignatureProviders = new ConcurrentDictionary<string, SignatureProvider>();
+        internal TokenCacheOptions _tokenCacheOptions;
+#if NET461 || NET472 || NETSTANDARD2_0 
+        private MemoryCache _signingSignatureProviders;
```

Comment:
_signingSignatureProviders and _verifyingSignatureProviders should be readonly in both targets

Reply:
Fixed!